stablehlo/dialect/StablehloOps.cpp

```diff
@@ -2294,148 +2294,112 @@ LogicalResult AllGatherOp::verify() {
-                            " but the size of scale factor is "
-                         << scaleShape << " and the feature count is "
-                         << featureCount << ".";
+  RankedTensorType operandType =
+      adaptor.operand().getType().cast<RankedTensorType>();
+  inferredReturnShapes.emplace_back(operandType.getShape(),
+                                    operandType.getElementType());
 
+  const int64_t featureCount = operandType.getShape()[adaptor.feature_index()];
+  SmallVector<int64_t> featureShape(1, featureCount);
+  inferredReturnShapes.emplace_back(featureShape, operandType.getElementType());
+  inferredReturnShapes.emplace_back(featureShape, operandType.getElementType());
   return success();
 }
 
 //===----------------------------------------------------------------------===//
 // BatchNormInferenceOp
 //===----------------------------------------------------------------------===//
 
+// Refer ODS for properties that are already enforced including shapes and
+// element types. This verifier includes additional checks.
 LogicalResult BatchNormInferenceOp::verify() {
-  // The following properties are already enforced by the ODS:
-  //  1. 'operand' and 'result' are ranked tensors.
-  //  2. 'scale', 'offset', 'mean', 'variance' are 1D tensors.
-  //  3. Types of 'operand' and 'result' matches.
-  //  4. Same shapes for 'scale', 'offset', 'mean', & 'variance'.
-
-  auto operandType = operand().getType().cast<RankedTensorType>();
-  if (static_cast<int64_t>(feature_index()) >= operandType.getRank())
-    return emitOpError() << "expects feature_index to be smaller "
-                            "than the rank of operand type; got feature_index "
-                         << feature_index() << ", and rank "
-                         << operandType.getRank() << ".";
-
-  if (static_cast<int64_t>(feature_index()) < 0)
-    return emitOpError() << "expects feature_index to be a "
-                         << "non-negative number, got "
-                         << static_cast<int64_t>(feature_index()) << ".";
-
-  // Note:A valid value of feature-index implies 'operand_type.getRank() >=1'.
-
-  const int64_t featureCount = operandType.getShape()[feature_index()];
-  const int64_t scaleSize =
-      scale().getType().cast<RankedTensorType>().getShape()[0];
-  // Check number of elements in input 'scale' equals feature_count.
-  // Together with (4) implies that 'scale', 'offset', 'mean', &
-  // 'variance' all have the same shape.
-  if (scaleSize != featureCount)
-    return emitOpError() << "expects the size of scale factor to be "
-                            "same as the feature count,"
-                            " but the size of scale factor is "
-                         << scaleSize << " and the feature count is "
-                         << featureCount << ".";
+  if (failed(verifyBatchNorm(getLoc(), operand(), feature_index(), scale())))
+    return failure();
```

Comment:
Are we checking that `operand`'s rank is at least 1? https://github.com/tensorflow/tensorflow/blob/d078f90b2833df402ff5c39e328f46e0bfd761b1/tensorflow/compiler/xla/service/shape_inference.cc#L1347-L1352

Reply:
same as the comment above.